### ISSUE TITLE
Uploading streams

### ISSUE THREAD

**Issue by ecoopnet** [ishkawa/APIKit#146]
I want to upload stream without creating NSData (like Alamofire/AFNetworking)
because uploading very large file(like long movie file) will cause crash by memory problem.

I tried to create custom RequestBodyBuilder but buildBodyFromObject return NSData...  
Please tell me how to fix this problem on APIKit 1 or 2?

**Comment by ishkawa** [comment]
APIKit 2 supports both NSData and NSInputStream, but APIKit 1 supports NSData only.

On APIKit 2, `RequestBodyBuilder ` enumeration is converted to `BodyParametersType` protocol, which returns `RequestBodyEntity`. Since `RequestBodyEntity` can represents `NSInputStream`, your custom body parameters can return `NSInputStream`.

```swift
public enum RequestBodyEntity {
    case Data(NSData)
    case InputStream(NSInputStream)
}

public protocol BodyParametersType {
    var contentType: String { get }
    func buildEntity() throws -> RequestBodyEntity
}
```

By the way, I will adopt `NSInputStream` in `MultipartFormDataParameters` in the future beta of APIKit 2.

**Comment by ecoopnet** [comment]
Thats great!

I am just using multipart form data on APIKit 1.
So I'll try APIKit2 next time.

Thanks for your nice answer.

**Comment by ishkawa** [comment]
`NSStream`-based `MultipartFormDataParameters` is implemented in #160 and available in 2.0.0 beta 4!
